org.eclipse.jdt.ui/ui/org/eclipse/jdt/ui/text/folding/DefaultJavaFoldingStructureProvider.java

```diff
@@ -979,10 +980,53 @@ private void computeFoldingStructure(FoldingStructureComputationContext ctx) {
 
 			ctx.getScanner().setSource(source.toCharArray());
 			computeFoldingStructure(parent.getChildren(), ctx);
+			addFoldingRegions(source, ctx);
 		} catch (JavaModelException x) {
 		}
 	}
 
+	private void addFoldingRegions(String source, FoldingStructureComputationContext ctx) {
```

Comment:
This name is not very descriptive. Rename to something like:
```suggestion
	private void computeFoldingStructureOfControlStatements(String source, FoldingStructureComputationContext ctx) {
```

Reply:
Done